### DISCUSSION TITLE
Going Prod

### DISCUSSION THREAD

**Discussion by DiegoDeMauro** [cloud-carbon-footprint/cloud-carbon-footprint#1284] [Q&A] [answered]
Can someone help?

After testing with good results, I would like to setup the Dashboard on a production environment, and was unable to find in the documentation how to tweek some parameters.
after clonning the git repository, configuring and starting, it seems to be starting a development webpack in port 3000. 
Changing the REACT_APP_BASE_URL in .env of the client did not took any effect aparently, and trying to access the dashboard using a dns name returns Invalid Host header.

Thanks

**Comment by 4upz** [comment, marked as answer]
Hi @DiegoDeMauro-Navigator The REACT_APP_BASE_URL refers to the base url of the api that your React app will be making requests to. This is why it didn't have the intended effect that you wanted it to have. 

If you wish to change the port that your React App is running on, you can view this [StackOverflow answer.](https://stackoverflow.com/questions/40714583/how-to-specify-a-port-to-run-a-create-react-app-based-project) 

However, I assume you are trying to deploy your application with a custom domain? If so, I don't believe changing the port is what you are looking for but rather how to map a custom domain to the server your CCF frontend is running on. It would be helpful to know a little bit more info on how/where you're trying to deploy the dashboard. Since we use Create-React-App under the hood, you're welcome to check out some of their [deployment docs ](https://stackoverflow.com/questions/40714583/how-to-specify-a-port-to-run-a-create-react-app-based-project)as well.

**Reply by DiegoDeMauro** [reply]
Thanks for the response, and yes, I´m trying to deploy it to a custom domain, to a ubuntu vm in azure, I was able to change the port for the client by adding PORT=80 to the .env, and you are absolutly right, port is not the important here, as I could have a app gateway doing the port wrap but would fall in the same result bellow (invalid host header) when using a custom dns name for the app gateway.

Following the same logic (sorry but I´m not react skilled at all), and reading the documentation, found that setting HOST, PUBLIC_URL to .env would do the trick, but that did not helped much, as Azure do not attach a internet IP directly to the virtual machine, so when starting using yarn start, if fails saying it could not open the port in the public dns Ip that I set (proably searching for open ports and the ip is not there).

I would suggest also to add to the project documentation, these types of configuration too, as I think most of the people interested in it would try to deploy it to their cloud, and access it using a dns name, not a IP.
